### PR TITLE
124/component/uip 137 popover

### PR DESCRIPTION
**Pull request by akdetrick**
## Description
Early feedback for `Popover`. Going to add a `Tooltip` and get some design review before merging.

## Screenshots

**run storybook locally, it's easier**

## Checklist
- [x] Docs have been rebuilt (`yarn build:full`)
- [x] All unit tests pass
- [x] Snapshots have been updated
- [x] No lint errors or warnings have been introduced
- [x] **[Version bumped](https://github.com/cbinsights/form-design-system#updating-version-number) if appropriate**

### CODE REVIEW

File: src/components/popovers/Popover.jsx [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+  const handleKeyPress = (e) => {
+    const isEscapeKey = ['Esc', 'Escape'].some((key) => key === e.key);
+    if (isEscapeKey) setIsActive(false);
+  };
+
+  /**
+   * Closes popover when user clicks outside of content or trigger
+   * @param {Event} e DOMEvent
+   */
+  const handleBodyMouseMove = (e) => {
```
Comment (akdetrick) [inline src/components/popovers/Popover.jsx]: 🎼 HANDLE BODY MOUSE MOVE 🎶

https://youtu.be/LaTGrV58wec

**Comment by rey-wright** [comment]
Screen Shot 2019-10-24 at 10 58 26 AM

I assumed that if the browser starts constraining the tooltip too much, then the tooltip would allow itself to get cut off. Instead, this tooltip aggressively wants to be visible, to the point where it's willing to completely overlap the Button. Do we think this is an issue?

**Comment by rey-wright** [comment]
I'm not really sure this distance prop is working as expected.... it doesn't really seem to do anything when I played with it.

**Comment by rey-wright** [comment]
Screen Shot 2019-10-24 at 11 31 17 AM

This could also just be me but I feel like the popover shouldn't overlap at all with the trigger... whereas here the red border is inside the bounds of the button, overlapping it slightly (though maybe you're still working on this part?)

**Comment by rey-wright** [comment]
Another thing: when the popover is in hovermode, clicking on the actual popover dismisses it. Do we want that? My assumption would be that once you hover over the trigger, you have to mouse out of both the trigger and the popover to dismiss it (else, things like menus and the like inside popover that you might want to click multiple times inside won't necessarily work).

File: src/components/popovers/Popover.jsx [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+  useEffect(() => {
+    setIsActive(interactionMode === 'controlled' && isOpen);
+  }, [interactionMode, isOpen]);
+
+  /**
+   * Closes popover when user presses ESC
+   * @param {Event} e DOMEvent
+   */
+  const handleKeyPress = (e) => {
+    const isEscapeKey = ['Esc', 'Escape'].some((key) => key === e.key);
```
[the remark this answers is not in the document]
Comment (akdetrick) [reply src/components/popovers/Popover.jsx]: IE/Edge = `Esc`
Normal good browsers that are normal and good = `Escape`

**Comment by akdetrick** [comment]
> Screen Shot 2019-10-24 at 10 58 26 AM
> 
> I assumed that if the browser starts constraining the tooltip too much, then the tooltip would allow itself to get cut off. Instead, this tooltip aggressively wants to be visible, to the point where it's willing to completely overlap the Button. Do we think this is an issue?

I think this is fine. The hover/focus/click modes all allow a way to dismiss the content even if the trigger is hidden.

**Comment by akdetrick** [comment]
@rey-wright transitions added - this is ready to merge after your review of the changes and a quick review of tooltip with Asaf.

**Comment by rey-wright** [comment]
Screen Shot 2019-10-31 at 2 59 21 PM

I guess the only problem I have with using the select knob, is 2 things. One, there's no way to show what the default value is (it could be that you selected the option that was the default, but that may not necessarily be the case). The 2nd, is that it makes seeing all the props at a glance that the Popover supports slightly more difficult.

**Comment by akdetrick** [comment]
changed position and interaction mode knobs to `inline-radio`

**Pull request merged**